Resources/Catalog/RucioFileCatalogClient.py

```diff
@@ -0,0 +1,590 @@
+                                                                                  'Mode': 436,
+                                                                                  'ModificationDate': datetime.datetime(1970, 1, 1),
+                                                                                  'GUID': str(file_dict.get(name, 'UNKNOWN')),
+                                                                                  'LastAccess': 0,
+                                                                                  'Size': rep['bytes']}
+              for rse in rep['rses']:
+                pfn = rep['rses'][rse][0]
+                if self.convertUnicode:
+                  pfn = str(pfn)
+                result['Value']['Successful'][lfn]['Files'][name]['Replicas'][rse] = {'PFN': pfn, 'Status': 'U'}
+      except DataIdentifierNotFound as err:
+        result['Value']['Failed'][lfn] = 'No such file or directory'
+      except Exception as err:
+        return S_ERROR(str(err))
+    return result
+
+
+  @checkCatalogArguments
+  def getFileMetadata(self, lfns, ownership = False):
+    """ Returns the file metadata associated to a supplied LFN
+    """
+    successful, failed = {}, {}
+    lfnChunks = breakListIntoChunks(lfns, 1000)
+    listFiles = deepcopy(list(lfns))
+    for chunk in lfnChunks:
+      try:
+        dids = [self.__getDidsFromLfn(lfn) for lfn in chunk]
+        for meta in self.client.get_metadata_bulk(dids):
+          lfn = str(meta['name'])
+          if meta['did_type'] in ['DATASET', 'CONTAINER']:
+            nlinks = len([child for child in self.client.list_content(meta['scope'], meta['name'])])
+            successful[lfn] = {'Checksum': '',
+                               'ChecksumType': '',
+                               'CreationDate': meta['created_at'],
+                               'GUID': '',
+                               'Mode': 509,
+                               'ModificationDate': meta['updated_at'],
+                               'NumberOfLinks': nlinks,
+                               'Size': 0,
+                               'Status': '-'}
+            try:
+              listFiles.remove(lfn)
+            except ValueError:
+              pass
+          else:
+            guid = meta['guid']
+            if guid:
+              guid = str(uuid.UUID(guid))
+            successful[lfn] = {'Checksum': str(meta['adler32']),
+                               'ChecksumType': 'AD',
+                               'CreationDate': meta['created_at'],
+                               'GUID': guid,
+                               'Mode': 436,
+                               'ModificationDate': meta['updated_at'],
+                               'NumberOfLinks': 1,
+                               'Size': meta['bytes'],
+                               'Status': '-'}
+            try:
+              listFiles.remove(lfn)
+            except ValueError:
+              pass
```

Comment:
These doesn't need to be duplicated as the same thing is done for both branches of the `if`